tensorboard/components/tf_backend/requestManager.ts

```diff
@@ -207,14 +210,16 @@ export class RequestManager {
   /* Actually get promise from url using XMLHttpRequest */
   protected _promiseFromUrl(url: string, requestOptions: RequestOptions) {
     return new Promise((resolve, reject) => {
-      const req = buildXMLHttpRequest(
-        requestOptions.methodType,
-        url,
-        requestOptions.withCredentials,
-        requestOptions.contentType);
+      const req = buildXMLHttpRequest(url, requestOptions);
       req.onload = function() {
         if (req.status === 200) {
-          resolve(JSON.parse(req.responseText));
+          // Edge and IE do not support "responseType == 'json'".
+          // Fall back to manual parsing.
+          if (req.responseType == 'json' && typeof req.response == 'string') {
```

Comment:
Won’t this try to double-parse if the result is an actual JSON string?

Reply:
In case the result is an actual JSON string, the `response` will be an object (in non-IE case)

Reply:
Hmm, doesn’t look like that to me—

```javascript
fetch("./test.json")
  .then((x) => x.json())
  .then((r) => void console.log("from fetch: %s of type %s", r, typeof r));

var xhr = new XMLHttpRequest();
xhr.open("GET", "./test.json", true);
xhr.responseType = "json";
xhr.onload = () => {
  var r = xhr.response;
  console.log("from XHR: %s of type %s", r, typeof r);
};
xhr.send();
```

```
from fetch: hello of type string
from XHR: hello of type string
```

(The server is `printf '"hello"' >test.json && python3 -m http.server`.
Client is Chrome M73 on Linux.)